kong/plugins/api-response-merger/body_transformer.lua

```diff
@@ -111,18 +138,18 @@ local function fetch(resource_key, resource_config, http_config)
   timer:stop()
   if not res then
     kong.log.err('Invalid response from upstream resource url: ', req_uri , ' err: ', err)
-    return {false, err}
+    return {false, create_error_response('invalid response from resource', req_uri, err)}
   end
 
   if res.status ~= 200 then
```

Comment:
Let me get it, so if you get other response than 200 (e.g. 404, because one resource is not there), you will always return message with 500? And you won't try to pass any meaningful error to the FE, the just need to look for it in the errors list? 
I don't like, but this is obviously FE problem, not mine :(

Reply:
Yes. This code is responsible for handling response from resources. Because we have a flat structure of data I don't have place where can I put error. For example:
I have upstream with a response:
```json 
{
"serviceA" : {"id": "id-a"}
}
```
If I ask serviceA and it will return 404, what should I do? Where to put the error?